<launch>
  <arg name="gui" default="true" />
  <arg name="GPU" default="0" />
  <arg name="INPUT_IMAGE" default="/hsrb/head_rgbd_sensor/rgb/image_rect_color" />

  <arg name="machine" default="tx1" />

  <include file="$(find jsk_hsr_startup)/jsk_hsr.machine" />

  <node name="people_pose_estimation_2d"
        pkg="jsk_perception" type="people_pose_estimation_2d.py"
        machine="$(arg machine)" output="screen">
    <remap from="~input" to="$(arg INPUT_IMAGE)" />
    <!--remap from="~input/info" to="$(arg INPUT_CAMERA_INFO)" /-->
    <!--remap from="~input/depth" to="$(arg INPUT_DEPTH_IMAGE)" /-->
    <rosparam subst_value="true">
      gpu: $(arg GPU)
      model_file: /etc/ros/jsk_data/jsk_perception/pose_estimation_2d_chainermodel.pkl
      with_depth: false
      scales: [0.38]
      stride: 8
    </rosparam>
  </node>

  <node name="facepose_detector" pkg="roboenvcv"
        type="facepose_detector.py" output="screen"
        machine="$(arg machine)">
      <param name="gpu" value="true"/>
  </node>

</launch>
